src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -191,34 +184,27 @@ bool SystemDictionary::is_platform_class_loader(oop class_loader) {
 }
 
 Handle SystemDictionary::compute_loader_lock_object(Thread* thread, Handle class_loader) {
-  // If class_loader is NULL we synchronize on _system_loader_lock_obj
-  if (class_loader.is_null()) {
-    return Handle(thread, _system_loader_lock_obj.resolve());
+  // If class_loader is NULL or parallelCapable, the JVM doesn't acquire a lock while loading.
+  if (is_parallelCapable(class_loader)) {
+    return Handle();
   } else {
     return class_loader;
   }
 }
 
 // This method is added to check how often we have to wait to grab loader
 // lock. The results are being recorded in the performance counters defined in
-// ClassLoader::_sync_systemLoaderLockContentionRate and
 // ClassLoader::_sync_nonSystemLoaderLockContentionRate.
 void SystemDictionary::check_loader_lock_contention(Thread* thread, Handle loader_lock) {
   if (!UsePerfData) {
     return;
   }
 
-  assert(!loader_lock.is_null(), "NULL lock object");
+  if (loader_lock.is_null()) return;
 
   if (ObjectSynchronizer::query_lock_ownership(thread->as_Java_thread(), loader_lock)
       == ObjectSynchronizer::owner_other) {
-    // contention will likely happen, so increment the corresponding
-    // contention counter.
-    if (loader_lock() == _system_loader_lock_obj.resolve()) {
-      ClassLoader::sync_systemLoaderLockContentionRate()->inc();
-    } else {
-      ClassLoader::sync_nonSystemLoaderLockContentionRate()->inc();
-    }
+    ClassLoader::sync_nonSystemLoaderLockContentionRate()->inc();
   }
```

Comment:
Should the systemLoaderLockContentionRate perf counter be removed as well? I don't know what kind of compatibility risk is there. Maybe we can doit in a separate RFE.

Reply:
So I think it should be done separately.  It was always zero or close to zero since nothing really contended on this lock.  I was wondering if any of these class loading perf counters are used by anything.  Something I wanted to look into afterwards.